### PR TITLE
Restrict Java 8 runtime forcing to compat projects

### PR DESCRIPTION
**Pull request by meltsufin**

### CODE REVIEW

**Comment by loosebazooka** [comment]
appveyor change needs to go in first?

File: src/main/java/com/google/cloud/tools/maven/StageMojo.java [hunk 1]
```diff
@@ -297,4 +309,45 @@ public File getDockerDirectory() {
   public String getRuntime() {
     return runtime;
   }
+
+
+  /**
+   * Simple parser for appengine-web.xml, this should ideally not exist, but we need it to correctly
+   * error when vm=false and the user is using java8 as the target platform
+   */
+  private static class AppEngineWebXml {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/StageMojo.java]: I guess this is choice, but would have been nice to have this non-nested-class?

**Comment by loosebazooka** [comment]
:runner:

File: src/main/java/com/google/cloud/tools/maven/StageMojo.java (hunk 1 above)
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/maven/StageMojo.java]: Yes, but ultimately, if this is going to stay around, I would like it to be in the lib.

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/StageMojo.java]: good point

**Pull request merged**